omnigibson/maps/traversable_map.py

```diff
@@ -182,40 +107,64 @@ def n_floors(self):
         """
         return len(self.floor_heights)
 
-    def get_random_point(self, floor=None):
+    def get_random_point(self, floor=None, prev_point=None, robot=None):
         """
         Sample a random point on the given floor number. If not given, sample a random floor number.
+        If @prev_point is given, sample a point in the same connected component as the previous point.
 
         Args:
             floor (None or int): floor number. None means the floor is randomly sampled
+            prev_point (None or 2-tuple): if given, sample a point in the same connected component as the previous point
+                - int: floor number of previous point
+                - 3-array: (x,y,z) previous point
 
         Returns:
             2-tuple:
                 - int: floor number. This is the sampled floor number if @floor is None
                 - 3-array: (x,y,z) randomly sampled point
         """
-        if floor is None:
+
+        # If the given floor and prev_point are not on the same floor, raise an error
+        if floor and prev_point and floor != prev_point[0]:
+            raise ValueError("floor and prev_point are not on the same floor")
+        # If nothing is given, sample a random floor and a random point on that floor
+        if floor is None and prev_point is None:
             floor = np.random.randint(0, self.n_floors)
-        trav = self.floor_map[floor]
-        trav_space = np.where(trav == 255)
+        
+        # create a deep copy so that we don't erode the original map
+        trav_map = copy.deepcopy(self.floor_map[floor])
+        
+        # Erode the traversability map to account for the robot's size
+        if robot:
+            if hasattr(robot, 'tucked_aabb_extent'):
```

Comment:
maybe we can make this more elegant?

use a single property, e.g.: `base_aabb_extent` (base referring to the physical robot locomotive base, not base class), and by default this is the pre-computed, static, cached aabb extent specified in the robot's frame.

And then for specific robots that has tucking capabilities (Fetch, Tiago), their robot class can extend the method to account for tucking.

This way we:
- only have a singled, unified property
- never have to actually compute `aabb_extent` on the fly
- the extent will be guaranteed to be consistent across calls for all robot classes

What do you think?

Reply:
lets discuss this in person quickly tomorrow. there's no perfect solution here and I'm hesitant to spend more time on this - but I do agree that the current approach is not great either so we should discuss

Reply:
As suggested by @cremebrule and what we talked about during our meeting, I've updated this with a single unified property `base_aabb_extent`. Please check the recent commits for the updated code.